src/main/java/com/kruize/service/RecommendationsService.java

```diff
@@ -117,25 +118,45 @@ private JsonObject getApplicationJson(ApplicationRecommendationsImpl application
 
         JsonObject resourcesJson = getResourceJson(applicationRecommendations, application);
 
-        applicationRecommendationJson.add("resources", resourcesJson);
-
-        return applicationRecommendationJson;
+        if (resourcesJson != null)
+        {
+            applicationRecommendationJson.add("resources", resourcesJson);
+            return applicationRecommendationJson;
+        }
+        else
+        {
+            return null;
+        }
     }
 
     private JsonObject getResourceJson(ApplicationRecommendationsImpl applicationRecommendations, String application) throws NoSuchApplicationException
     {
-        JsonObject resourcesJson = new JsonObject();
+        String applicationStatus = applicationRecommendations.getStatus(application);
 
-        JsonObject resourceRequestsJson = new JsonObject();
-        resourceRequestsJson.addProperty("memory", applicationRecommendations.getRssRequests(application) + "MB");
-        resourceRequestsJson.addProperty("cpu", applicationRecommendations.getCpuRequests(application));
+        /* If application is still running or idle, or if application is removed,
+            but Kruize has earlier generated recommendations.
+         */
+        if (applicationStatus.equals("running") || applicationStatus.equals("idle")
+                || (applicationRecommendations.getRssRequests(application) != 0))
```

Comment:
@chandrams Only if the application is running or idle, or it has been removed, but non zero recommendations were already generated, will the application recommendations be exported by Kruize through the API. Please let me know if this is OK

Reply:
That looks ok to me